src/pybind/mgr/rbd_support/mirror_snapshot_schedule.py

```diff
@@ -364,6 +365,97 @@ def run(self) -> None:
             self.log.fatal("Fatal runtime error: {}\n{}".format(
                 ex, traceback.format_exc()))
 
+    def get_queue_timestamp(
+        self,
+        pool_id: str,
+        namespace: str,
+        image_id: str,
+    ) -> Optional[datetime]:
+        for timestamp, images in self.queue.items():
+            for img in images:
+                if img[0] == pool_id and img[1] == namespace and img[2] == image_id:
+                    return datetime.strptime(timestamp, '%Y-%m-%d %H:%M:%S')
+        return None
+    
+    def get_health_checks(self) -> dict[str, any]:
+        unsynced_images = []
+        slow_status_images = []
+        total_image_count = 0
+
+        for pool_id in self.images:
+            with self.module.rados.open_ioctx2(int(pool_id)) as ioctx:
+                for namespace in self.images[pool_id]:
+                    ioctx.set_namespace(namespace)
+                    try:
+                        statuses = list(rbd.RBD().mirror_image_status_list(ioctx))
+                    except Exception as e:
+                        self.log.error(
+                            "error getting image statuses from {}/{}: {}".format(
+                                pool_id, namespace, e
+                            )
+                        )
+                        continue
+
+                    for image_id in self.images[pool_id][namespace]:
+                        schedule = self.schedules.find(pool_id, namespace, image_id)
+                        if not schedule:
+                            continue
+                        total_image_count += 1
+                        timestamps = get_image_timestamps(statuses, image_id)
+                        if not timestamps:
+                            continue
```

Comment:
Since images that don't have a status are silently skipped here, why call `get_image_timestamps` for each image, making this quadratic in time?  Could the list of statuses be traversed instead of `self.images`?